x-pack/legacy/plugins/siem/scripts/convert_saved_search_to_signals.js

```diff
@@ -0,0 +1,131 @@
+const INDEX = ['auditbeat-*', 'filebeat-*', 'packetbeat-*', 'winlogbeat-*'];
+
+const walk = dir => {
+  const list = fs.readdirSync(dir);
+  return list.reduce((accum, file) => {
+    const fileWithDir = dir + '/' + file;
+    const stat = fs.statSync(fileWithDir);
+    if (stat && stat.isDirectory()) {
+      return [...accum, ...walk(fileWithDir)];
+    } else {
+      return [...accum, fileWithDir];
+    }
+  }, []);
+};
+
+// Temporary hash function for converting string to numbers.
+// TODO: Once we move from numbers to pure strings for id's this can be removed
+// and the file name used as the id (or a GUID), etc...
+const hashFunc = str => {
+  let chr;
+  let hash = 0;
+  if (str.length === 0) return 0;
+  for (let i = 0; i < str.length; i++) {
+    chr = str.charCodeAt(i);
+    // eslint-disable-next-line no-bitwise
+    hash = (hash << 5) - hash + chr;
+    // eslint-disable-next-line no-bitwise
+    hash |= 0;
+  }
+  return hash;
+};
+
+//clean up the file system characters
+const cleanupFileName = file => {
+  return path
+    .basename(file, path.extname(file))
+    .replace(/\s+/g, '_')
+    .replace(/,/g, '')
+    .replace(/\+s/g, '')
+    .replace(/-/g, '')
+    .replace(/__/g, '_')
+    .toLowerCase();
+};
+
+async function main() {
+  if (process.argv.length !== 4) {
+    throw new Error(
+      'usage: saved_query_to_signals [input directory with saved searches] [output directory]'
+    );
+  }
+
+  const files = process.argv[2];
+  const outputDir = process.argv[3];
+
+  const savedSearchesJson = walk(files).filter(file => file.endsWith('.ndjson'));
+
+  const savedSearchesParsed = savedSearchesJson.reduce((accum, json) => {
+    const jsonFile = fs.readFileSync(json, 'utf8');
+    try {
+      const parsedFile = JSON.parse(jsonFile);
+      parsedFile._file = json;
+      parsedFile.attributes.kibanaSavedObjectMeta.searchSourceJSON = JSON.parse(
+        parsedFile.attributes.kibanaSavedObjectMeta.searchSourceJSON
+      );
+      return [...accum, parsedFile];
+    } catch (err) {
+      return accum;
+    }
+  }, []);
+
+  savedSearchesParsed.forEach(savedSearch => {
+    const fileToWrite = cleanupFileName(savedSearch._file);
+
+    const query = savedSearch.attributes.kibanaSavedObjectMeta.searchSourceJSON.query.query;
+    if (query != null && query.trim() !== '') {
+      const outputMessage = {
+        id: `${hashFunc(fileToWrite)}`, // TODO: Remove this once we change id to a string
+        description: savedSearch.attributes.description || savedSearch.attributes.title,
+        index: INDEX,
+        interval: INTERVAL,
+        name: savedSearch.attributes.title,
+        severity: SEVERITY,
+        type: TYPE,
+        from: FROM,
+        to: TO,
+        kql: savedSearch.attributes.kibanaSavedObjectMeta.searchSourceJSON.query.query,
+      };
+
+      fs.writeFileSync(`${outputDir}/${fileToWrite}.json`, JSON.stringify(outputMessage, null, 2));
+    }
+  });
+}
+
+if (require.main === module) {
+  main();
+}
```

Comment:
Looking for a way to make files like this a TypeScript file. It is a command line script at the moment but we might end up having an endpoint which takes exported saved objects and just move most if not all of this logic into that endpoint eventually so maybe I should just concentrate on that instead.